src/jsutils/instanceOf.ts

```diff
@@ -8,8 +8,7 @@ import { inspect } from './inspect.js';
  */
 export const instanceOf: (value: unknown, constructor: Constructor) => boolean =
   /* c8 ignore next 6 */
-  // FIXME: https://github.com/graphql/graphql-js/issues/2317
-  globalThis.process?.env.NODE_ENV === 'production'
+  typeof process === 'undefined' || process.env.NODE_ENV === 'production'
```

Comment:
```suggestion
  typeof process !== 'undefined' && process.env.NODE_ENV === 'production'
```
This is a complete logic check otherwise.

Reply:
your change will result in running `dev` code for browsers that don't use bundlers

Reply:
Yes, that has always been the behavior of this check.